Objective-C/CBLReplicator.mm

```diff
@@ -597,7 +597,7 @@ - (void) _resolveConflict: (CBLReplicatedDocument*)doc {
                                withConflictResolver: _config.conflictResolver
                                               error: &error])
         [doc resetError];
-    else {
+    else if (error) {
```

Comment:
This means that if the error is nil, we have never reseted the error. I think we probably don't need the `resetError` method if we always update the error at the end of the `_resolveConflict:` method.

Reply:
then i feel like we should be returning success, else seems like we will end up with 


<img width="775" alt="Screen Shot 2019-06-19 at 10 14 43 AM" src="https://user-images.githubusercontent.com/10448770/59786002-1ad04400-927b-11e9-9f18-f020f14d6919.png">

Reply:
updated the change: https://github.com/couchbase/couchbase-lite-ios/pull/2515/commits/154b2e7240103a90d68086556f46fc3c282a705b